ddtrace/span.py

```diff
@@ -116,6 +112,17 @@ def start(self):
     def start(self, value):
         self.start_ns = int(value * 1e9)
 
+    @property
+    def finished(self):
+        return self.duration_ns is not None
+
+    @finished.setter
+    def finished(self, value):
+        if value is False:
+            self.duration_ns = None
+        elif value is True and not self.finished:
+            self.duration_ns = time_ns() - self.start_ns
+
```

Comment:
I'm not 100% about this implementation for the setter. Being able to mutate this attribute publicly doesn't really make sense in the first place. The only place we do it is to make testing some internals easier. We could instead deprecate setting the value publicly. Thoughts?

Reply:
I think it's a good idea and it's a nice way of finishing a trace without having to bother with the time function.

However, you currently ignore non boolean value and that's not good. Either:
- raise `ValueError` as last resort
- use `if value/else` and don't check type

Reply:
Good points. I did consider it but didn't think `span.finished = 342432` made much sense. The ValueError makes sense to me though!

Reply:
On second thought, I think the `if value/else` is a bit more pythonic. 🤷‍♂️